### PR TITLE
construct and assign portal and checkers consistently

### PR DESCRIPTION
**Pull request by graingert**
The example uses a mix of constructor form and the register form.

Use only the constructor form, and create all class properties in the classdef

## Contributor Checklist:

* [x] There is an associated ticket in Trac. Create a new one at https://twistedmatrix.com/trac/newticket
* [ ] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
* [ ] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [ ] I have updated the automated tests.

### CODE REVIEW

File: docs/conch/examples/sshsimpleserver.py [hunk 1]
```diff
@@ -224,15 +229,6 @@ def getPrimes(self):
...
 
 
-portal = portal.Portal(ExampleRealm())
-passwdDB = InMemoryUsernamePasswordDatabaseDontUse()
-passwdDB.addUser(b'user', b'password')
-sshDB = SSHPublicKeyChecker(InMemorySSHKeyDB(
-    {b'user': [keys.Key.fromFile(CLIENT_RSA_PUBLIC)]}))
-portal.registerChecker(passwdDB)
-portal.registerChecker(sshDB)
-ExampleFactory.portal = portal
```
Comment (graingert) [inline docs/conch/examples/sshsimpleserver.py:234]: when reading the code I was very confused about how ExampleFactory knew about the ExampleRealm and the various checkers.

IMHOP it's very unusual to define class properties outside of the classdef, so I missed this line completely

**Comment by cdunklau** [comment]
Good catch, this example is certainly confusing, and shows some probably-bad practice what with the assigning to class attributes and doing stuff at import time... but it looks like your changes basically just shuffle this around. I think you can do better :)

Instead of assigning to class attributes at all, you could change all of that factory tweaking to just tweak an *instance* of the factory instead of the class itself. This would best be done under the `if __name__ == '__main__':` branch, or better yet in a `def main():` that it calls.

**Comment by graingert** [comment]
You're right. factory class instances should be independent

I think this should be in the constructor

**Comment by cdunklau** [comment]
Oh yeah, the newsfile should probably be a .doc, not a .misc

**Comment by graingert** [comment]
closed in favor of https://github.com/twisted/twisted/pull/1096 for better branch name

**Pull request closed without merging**